pints/__init__.py

```diff
@@ -103,6 +103,11 @@ def version(formatted=False):
 #
 import pints.noise
 
+#
+# Toy pdfs
+#
+from .toy._logpdfs import LogPDF
```

Comment:
Why we want to separate `LogPDF` and `LogLikelihood`?

I think they are really the same thing when we use them. We can have all toy PDFs inherit the `LogLikelihood` class or `LogPosterior` (if we want to have it looks more "meaningful" -- as effectively, or eventually _we pretend_, they are the log posteriors, right?).

Reply:
I think this is resolved now, with Ben's comments below.

However, I'd like to add that I don't see why this is imported here! All other toy stuff isn't imported as part of pints proper, which is good, because we should only use stuff from the toy module when benchmarking / demoing / testing!

Reply:
@chonlei I just had a chat with Michael about this. let me explain.
In Bayesian inference we have:
logPosterior =LogLikelihood + LogPrior

Now for time series problems we calculate LogLikelihood, LogPrior and add them up to evaluate LogPosterior. But for some toy densities such as the Banana one we have a direct analytical form of the logPDF (not to confuse with posterior as in Bayesian sense). Thus
logPDF= f(Theta), f some function that we know in closed form. In this case the notion of likelihood and prior does not exist. MCMC is a general algorithm to sample from any unnormalised density that we can evaluate. Bayesian inference is one example problem that can be solved using MCMC, but is not limited to Bayesian only. When @ben18785  wants to sample from these toy distributions then he is not doing Bayesian inference, he is just sampling some known distributions that are hard to sample from.

Reply:
Yes! Or go even further and just let it be any kind of distribution? Or function?

Reply:
Good to keep its requirement for `dimension(self)` and `__call__(self, x)` though...

Reply:
And it makes sense to force the sampler to do sampling from an unnormalised PDF (as @sanmitraghosh said). Then it probably also make sense to have classes extend from the `logPDF (unnormalised)` as above, and indeed logPosterior, logPrior etc. belong to this category.